addons/purchase_stock/models/purchase.py

```diff
@@ -222,6 +222,27 @@ class PurchaseOrderLine(models.Model):
     orderpoint_id = fields.Many2one('stock.warehouse.orderpoint', 'Orderpoint')
     move_dest_ids = fields.One2many('stock.move', 'created_purchase_line_id', 'Downstream Moves')
 
+    @api.onchange('product_qty')
+    def _onchange_product_qty(self):
+        # When modifying a one2many, _origin doesn't guarantee that its values will be the ones
+        # in database. Hence, we need to explicitly read them from there.
+        if self._origin:
+            product_qty_origin = self._origin.read(["product_qty"])[0]["product_qty"]
+        else:
+            product_qty_origin = 0
+
+        if self.state == 'purchase' and self.product_id.type in ['product', 'consu'] and self.product_qty < product_qty_origin:
+            # Do not display this warning if the new quantity is below the received
+            # one; the `write` will raise an `UserError` anyway.
+            if self.product_qty < self.qty_received:
+                return {}
+            warning_mess = {
+                'title': _('Ordered quantity decreased!'),
+                'message' : _('You are decreasing the ordered quantity! Do not forget to manually update the receipt order if needed.'),
+            }
+            return {'warning': warning_mess}
+        return {}
+    
```

Comment:
Do you really need that?

Reply:
When decreasing the quantity ordered, the “stock.move” in picking is not updated, so it makes sense to notify the user with this message as is the case for SO, what do you think?

Reply:
I would not add it in the bugfix. It still an exception and it's not a blocking issue